### PR TITLE
PIA-882: Login feature for tvOS

### PR DESCRIPTION
**Pull request by kp-said-rehouni**
### Summary
This PR adds the Login screen to PIA tvOS. It includes all the different layers involved:
- UI
- Data
- Presentation
- Domain/App logic
- Composition Root

`LoginViewModel` exposes a status property in order to validate the logic when unit testing since the screen does not navigate when the operations was successful. This property will be taken care of once we tackle the navigation to the next screen. 

Tests are composed mainly by unit tests. There are also a couple of integration tests that are mocking the login provider which is a bridge to PIALibrary's `DefaultAccountProvider`. These tests are responsible of testing the integration between the all the different components on presentation and domain layers.

Error handling and proper design will be tackled in a different PR.

### Architecture

Screenshot 2023-12-11 at 22 23 30

### CODE REVIEW

File: PIA VPN-tvOS/Login/Domain/Interfaces/LoginProviderType.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//
+//  LoginProviderType.swift
+//  PIA VPN-tvOS
+//
+//  Created by Said Rehouni on 27/11/23.
+//  Copyright © 2023 Private Internet Access Inc. All rights reserved.
+//
+
+import Foundation
+import PIALibrary
```
Comment (kp-juan-docal) [inline PIA VPN-tvOS/Login/Domain/Interfaces/LoginProviderType.swift:10]: wouldn't this mean we are exposing `Data` specifics to `Domain`?

File: PIA VPN-tvOS/Login/Domain/Use cases/LoginWithCredentialsUseCase.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
+//
+//  LoginWithCredentialsUseCase.swift
+//  PIA VPN-tvOS
+//
+//  Created by Said Rehouni on 27/11/23.
+//  Copyright © 2023 Private Internet Access Inc. All rights reserved.
+//
+
+import Foundation
+import PIALibrary
```
Comment (kp-juan-docal) [inline PIA VPN-tvOS/Login/Domain/Use cases/LoginWithCredentialsUseCase.swift:10]: Same as above.

File: PIA VPN-tvOS/Login/Presentation/CheckLoginAvailability.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
+//
+//  CheckLoginAvailability.swift
+//  PIA VPN-tvOS
+//
+//  Created by Said Rehouni on 29/11/23.
+//  Copyright © 2023 Private Internet Access Inc. All rights reserved.
+//
+
+import Foundation
+import PIALibrary
```
Comment (kp-juan-docal) [inline PIA VPN-tvOS/Login/Presentation/CheckLoginAvailability.swift:10]: This is `Presentation`. So, we have essentially its dependency on all layers.

File: PIA VPN-tvOS/Login/Presentation/LoginPresentableErrorMapper.swift [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by Said Rehouni on 28/11/23.
+//  Copyright © 2023 Private Internet Access Inc. All rights reserved.
+//
+
+import Foundation
+
+class LoginPresentableErrorMapper {
+    func map(error: LoginError) -> String? {
+        return nil
```
Comment (kp-juan-docal) [inline PIA VPN-tvOS/Login/Presentation/LoginPresentableErrorMapper.swift:13]: is it temp that we are always returning `nil`?

Comment (kp-said-rehouni) [reply PIA VPN-tvOS/Login/Presentation/LoginPresentableErrorMapper.swift:13]: Yes exactly, we have a ticket for error handling where we will tackle all the different errors we might have

File: PIA VPN-tvOS/Login/Domain/Interfaces/LoginProviderType.swift (hunk 1 above)
Comment (kp-said-rehouni) [reply PIA VPN-tvOS/Login/Domain/Interfaces/LoginProviderType.swift:10]: You are right @kp-juan-docal!
There are some types in PIALibrary and some utility extensions we are using that's why I imported PIALibrary, but I agree it's better to not decouple PIALibrary with the rest of the layer. Will update this together with all the other comments you left.
Thank you!

**Review by kp-juan-docal — COMMENTED** [review]
There are still some `import PIALibrary` in the `Presentation` layer. Are they still needed?

File: PIA VPN-tvOS/Login/Presentation/CheckLoginAvailability.swift (hunk 3 above)
Comment (kp-said-rehouni) [reply PIA VPN-tvOS/Login/Presentation/CheckLoginAvailability.swift:10]: Good catch @kp-juan-docal! It's updated now 😅
Thank you!

**Pull request merged**